### PR TITLE
[Feature] Generated docs add work experience fields

### LINKED ISSUE #11869: ✨ Generated files - New work experience fields
## ✨ Feature

Add new the work experience fields to the user generated export files. Experience to not show up in CSV, so no need to update CSV.

## 🎨 Design

[File to google doc example](https://docs.google.com/document/d/1tKlZ381GBaT7eeJ3vAueTiw4MmzkVqQFWCn_b3W_dqU/edit?usp=sharing) by @Jerryescandon 

## ✅ Acceptance Criteria

- [ ] DOCX files match provided design

## 🛑 Blockers

```[tasklist]
### Blocked By
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/11868
```

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #11869

## 👋 Introduction

Update the profile download per [link](https://docs.google.com/document/d/1tKlZ381GBaT7eeJ3vAueTiw4MmzkVqQFWCn_b3W_dqU/edit?usp=sharing)

## 🕵️ Details

Whole lot of conditionals but I think rendering this effectively markup is going to look like that

## 🧪 Testing

1. Download various user profile docs
2. Scroll to work experiences and observe
3. Compare to example linked above

## 📸 Screenshot

image

image

image

image

### CODE REVIEW

**Comment by vd1992** [comment]
Now in French too! https://github.com/GCTC-NTGC/gc-digital-talent/pull/12108/commits/811691dfe4bd4ef683becca2d50050f50d7e2eae

**Review by petertgiles — CHANGES_REQUESTED** [review]
Works great!  Two small suggestions.

File: api/app/Models/WorkExperience.php [hunk 1]
```diff
@@ -88,6 +88,28 @@ public function getTitle(?string $lang = 'en'): string
...
+    public function getDateRangeWithFutureEndDateCheck($lang = 'en'): string
+    {
+        $format = 'MMM Y';
+        $start = $this->start_date->locale($lang)->isoFormat($format);
+        $now = date('Y-m-d');
+
+        if (isset($this->end_date)) {
+            $end = $this->end_date->locale($lang)->isoFormat($format);
+
+            if ($this->end_date > $now) {
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/app/Models/WorkExperience.php:102]: Yeah sure, https://github.com/GCTC-NTGC/gc-digital-talent/pull/12108/commits/ddaf14a1feb2a54bc336f628d403ed378674fdf8

File: api/app/Traits/Generator/GeneratesUserDoc.php [hunk 2]
```diff
@@ -298,15 +314,114 @@ public function experience(Section $section, AwardExperience|CommunityExperience
...
+                            $this->localize('experiences.contract_firm_agency'),
+                            $experience->contractor_firm_agency_name
+                        );
+                    }
+                }
+                if (
+                    $experience->gov_employment_type !== WorkExperienceGovEmployeeType::CONTRACTOR->name &&
+                    $experience->gov_employment_type !== WorkExperienceGovEmployeeType::STUDENT->name
+                ) {
+                    $classification = Classification::findOrFail($experience->classification_id);
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/app/Traits/Generator/GeneratesUserDoc.php]: That makes sense to me, yeah changed in https://github.com/GCTC-NTGC/gc-digital-talent/pull/12108/commits/ddaf14a1feb2a54bc336f628d403ed378674fdf8

**Review by petertgiles — APPROVED** [review]

**Pull request merged**